streams/src/test/java/org/apache/kafka/streams/state/internals/TimeOrderedKeyValueBufferTest.java

```diff
@@ -406,118 +411,18 @@ public void shouldRestoreOldFormat() {
-                                 -1L,
-                                 -1,
-                                 -1,
-                                 "asdf".getBytes(UTF_8),
-                                 ByteBuffer.allocate(Long.BYTES + asdfValue.length).putLong(2L).put(asdfValue).array(),
-                                 v1FlagHeaders),
-            new ConsumerRecord<>("changelog-topic",
-                                 0,
-                                 2,
-                                 99,
-                                 TimestampType.CREATE_TIME,
-                                 -1L,
-                                 -1,
-                                 -1,
-                                 "zxcv".getBytes(UTF_8),
-                                 ByteBuffer.allocate(Long.BYTES + zxcvValue.length).putLong(1L).put(zxcvValue).array(),
-                                 v1FlagHeaders)
-        ));
-
-        assertThat(buffer.numRecords(), is(3));
-        assertThat(buffer.minTimestamp(), is(0L));
-        assertThat(buffer.bufferSize(), is(130L));
-
-        stateRestoreCallback.restoreBatch(singletonList(
-            new ConsumerRecord<>("changelog-topic",
-                                 0,
-                                 3,
-                                 3,
-                                 TimestampType.CREATE_TIME,
-                                 -1L,
-                                 -1,
-                                 -1,
-                                 "todelete".getBytes(UTF_8),
-                                 null)
-        ));
-
-        assertThat(buffer.numRecords(), is(2));
-        assertThat(buffer.minTimestamp(), is(1L));
-        assertThat(buffer.bufferSize(), is(83L));
-
-        // flush the buffer into a list in buffer order so we can make assertions about the contents.
-
-        final List<Eviction<String, String>> evicted = new LinkedList<>();
-        buffer.evictWhile(() -> true, evicted::add);
-
-        // Several things to note:
-        // * The buffered records are ordered according to their buffer time (serialized in the value of the changelog)
-        // * The record timestamps are properly restored, and not conflated with the record's buffer time.
-        // * The keys and values are properly restored
-        // * The record topic is set to the original input topic, *not* the changelog topic
-        // * The record offset preserves the original input record's offset, *not* the offset of the changelog record
-
-
-        assertThat(evicted, is(asList(
-            new Eviction<>(
-                "zxcv",
-                "3o4im",
-                getContext(2L)),
-            new Eviction<>(
-                "asdf",
-                "qwer",
-                getContext(1L)
-            ))));
-
-        cleanup(context, buffer);
-    }
-
-    @Test
-    public void shouldNotRestoreUnrecognizedVersionRecord() {
+    public void shouldIgnoreHeadersOnRestore() {
```

Comment:
Do we need this test? Seems we could remove it?

Reply:
We could remove it, but having the test helps to reinforce that this behavior is in fact intentional.